torchmetrics/functional/text/bert.py

```diff
@@ -11,105 +11,667 @@
+
+    Raises:
+        ValueError:
+    """
+    EMBEDDINGS_LIST: List[torch.Tensor] = []
+    IDF_SCALE_LIST: List[torch.Tensor] = []
+    for batch in _get_progress_bar(dataloader, verbose):
+        with torch.no_grad():
+            batch = _input_data_collator(batch, device)
+            if not all_layers:
+                # Output shape: batch_size x 1 x sequence_length x bert_dim
+                if not user_forward_fn:
+                    out = (
+                        model(
+                            batch["input_ids"],
+                            batch["attention_mask"],
+                            output_hidden_states=True,
+                        )
+                        .hidden_states[num_layers if num_layers is not None else -1]
+                        .unsqueeze(1)
+                    )
+                else:
+                    out = user_forward_fn(model, batch)
+                    bs, seq_len = batch["input_ids"].shape[:2]
+                    invalid_out_shape = len(out.shape) != 3 or out.shape[0] != bs or out.shape[1] != seq_len
+                    if invalid_out_shape:
+                        raise ValueError(
+                            "The model output must be `torch.Tensor` of a shape `[batch_size, seq_len, model_dim]` "
+                            f"i.e. [{batch['input_ids'].shape[0]}, {batch['input_ids'].shape[0]}. , `model_dim`], "
+                            f"but got {out.shape}."
+                        )
+                    out = out.unsqueeze(1)
+            else:
+                if user_forward_fn:
+                    raise ValueError("")
+                # Output shape: batch_size x num_layers x sequence_length x bert_dim
+                out = model(batch["input_ids"], batch["attention_mask"], output_hidden_states=True).hidden_states
+                out = torch.cat([o.unsqueeze(1) for o in out], dim=1)
+
+        out /= out.norm(dim=-1).unsqueeze(-1)  # normalize embeddings
+        out, attention_mask = _output_data_collator(out, batch["attention_mask"], target_len)
+        processed_attention_mask = _process_attention_mask_for_special_tokens(attention_mask)
+        # Multiply embeddings with attention_mask (b=batch_size, l=num_layers, s=seq_len, d=emb_dim)
+        out = torch.einsum("blsd, bs -> blsd", out, processed_attention_mask)
+        EMBEDDINGS_LIST.append(out.cpu())
+
+        # Calculate weighted (w.r.t. sentence length) input_ids IDF matrix
+        input_ids_idf = (
+            batch["input_ids_idf"] * processed_attention_mask if idf else processed_attention_mask.type(out.dtype)
+        )
+        input_ids_idf /= input_ids_idf.sum(-1, keepdim=True)
+        IDF_SCALE_LIST.append(input_ids_idf)
+
+    EMBEDDINGS = torch.cat(EMBEDDINGS_LIST)
+    IDF_SCALE = torch.cat(IDF_SCALE_LIST)
+
+    return EMBEDDINGS, IDF_SCALE
+
+
+def _get_scaled_precision_or_recall(cos_sim: torch.Tensor, metric: str, idf_scale: torch.Tensor) -> torch.Tensor:
```

Comment:
```suggestion
def _get_scaled_precision_or_recall(cos_sim: Tensor, metric: str, idf_scale: Tensor) -> Tensor:
```
lets simplify if and do `from torch import Tensor`